syntax = "proto3";

package blog;

service BlogService {
  rpc CreatePost (CreatePostRequest) returns (Post);
  rpc GetPost (GetPostRequest) returns (Post);
  rpc ListPosts (ListPostsRequest) returns (ListPostsResponse);
}

message Post {
  string id = 1;
  string title = 2;
  string content = 3;
  string author = 4;
  string created_at = 5;
}

message CreatePostRequest {
  string title = 1;
  string content = 2;
  string author = 3;
}

message GetPostRequest {
  string id = 1;
}

message ListPostsRequest {
  int32 page = 1;
  int32 page_size = 2;
}

message ListPostsResponse {
  repeated Post posts = 1;
  int32 total = 2;
}